Hyeonwoo-Park/BOJ17144/Dust.java

```diff
@@ -0,0 +1,117 @@
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.StringTokenizer;
+
+public class Dust {
+
+	private static int dx[] = { 0, 1, 0, -1 };
+	private static int dy[] = { 1, 0, -1, 0 };
+
+	static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+	static int R, C, T;
+	static int[][] room, purifiers, temp;
+
+	private static void defuse(int x, int y, int dust) {
+		for (int i = 0; i < 4; ++i) {
+			int X = x + dx[i];
+			int Y = y + dy[i];
+
+			if (X < 0 || Y < 0 || X >= R || Y >= C)
+				continue;
+			if (room[X][Y] == -1)
+				continue;
+
+			if (i < 2)
+				temp[X][Y] += dust;
+			else
+				room[X][Y] += dust;
+
+			room[x][y] -= dust;
+		}
+	}
+
+	private static void rotate(int purifiers[][]) {
+
+		for (int i = 0; i < 2; ++i) {
+			int x = purifiers[i][0];
+			int y = purifiers[i][1];
+			int dir = 2;
+			int tb = i == 0 ? 1 : -1;
+
+			int ceil = i == 0 ? 0 : x;
+			int floor = i == 0 ? x + 1 : R;
+
+			while (true) {
+				int nx = x + dx[dir];
+				int ny = y + dy[dir];
+				
+				if (nx < ceil || ny < 0 || nx >= floor || ny >= C) {
+					dir += tb;
+
+					if (dir == -1)
+						dir = 3;
+
+					dir %= 4;
```

Comment:
```suggestion
					dir = (dir+4)%4;
```
간단하게는 이렇게도 표현할 수 있어요~!